libraries/botbuilder-dialogs-declarative/tests/schemaMergeTest.js

```diff
@@ -0,0 +1,141 @@
+const { FolderResourceProvider, ResourceExplorer, FileResource } = require('../lib');
+const path = require('path');
+const assert = require('assert');
+const fs = require('fs');
+const Validator = require('jsonschema').Validator;
+// eslint-disable-next-line security/detect-child-process
+const { exec } = require('child_process');
+
+// Note: This file is intentionally not named *.test.js to ensure it isn't run
+// via `yarn run test` or `npm run test`.
+
+async function runCommand(command) {
+    try {
+        return new Promise((resolve, _reject) => {
+            exec(
+                command,
+                {
+                    cwd: __dirname,
+                },
+                (err, stdout, stderr) => {
+                    if (err || stderr) {
+                        console.error(err ?? stderr);
+                    }
+                    resolve({ error: err ?? stderr, log: stdout });
+                }
+            );
+        });
+    } catch (error) {
+        // Errors *should* be handled in the callback above, but don't seem to be, so we'll try/catch as well.
+        return { error };
+    }
+}
+
+describe('Schema Merge Tests', function () {
+    const resourceExplorer = new ResourceExplorer();
+    const resourceProvider = new FolderResourceProvider(
+        resourceExplorer,
+        path.join(__dirname, '..', '..'),
+        true,
+        false
+    );
+    const dialogResources = resourceProvider.getResources('dialog');
+    const dialogs = dialogResources
+        .filter((r) => !r.id.endsWith('.schema.dialog'))
+        .map((resource) => {
+            return {
+                id: resource.id,
+                fullName: resource.fullName,
+            };
+        });
+
+    const testsSchemaPath = path.join(__dirname, '..', '..', 'tests.schema');
+    const testsSchemaFileResource = new FileResource(testsSchemaPath);
+    const testsSchema = JSON.parse(testsSchemaFileResource.readText());
+
+    // This fixture creates or updates test.schema by calling bf dialog:merge on all the schema files in the solution.
+    // This will install the latest version of botframewrork-cli if the schema changed and npm is present.
+    it('should generate a new tests.schema file, if necessary', async function () {
+        this.timeout(50000);
+        // This only runs on Windows platforms.
+        if (process.platform === 'win32') {
```

Comment:
Conditionally using `it.skip` is a better approach because it properly reports skipped tests to `mocha` rather than silently succeeding. [See example here](https://github.com/microsoft/botbuilder-js/blob/main/libraries/botbuilder-azure-blobs/tests/blobsStorage.test.js#L16).